src/ui/reducers/network.ts

```diff
@@ -0,0 +1,46 @@
+import { createSelector } from "reselect";
+import { RequestEventInfo, RequestInfo } from "@recordreplay/protocol";
+import { UIState } from "ui/state";
+
+export type NetworkState = {
+  events: RequestEventInfo[];
+  requests: RequestInfo[];
+};
+
+const initialState = (): NetworkState => ({
+  events: [],
+  requests: [],
+});
+
+type NewNetworkRequestsAction = {
+  type: "NEW_NETWORK_REQUESTS";
+  payload: { requests: RequestInfo[]; events: RequestEventInfo[] };
+};
+
+type NetworkAction = NewNetworkRequestsAction;
+
+const update = (state: NetworkState = initialState(), action: NetworkAction): NetworkState => {
+  switch (action.type) {
+    case "NEW_NETWORK_REQUESTS":
+      return {
+        events: [...action.payload.events, ...state.events],
+        requests: [...action.payload.requests, ...state.requests],
+      };
+    default:
+      return state;
+  }
+};
+
+const getEvents = (state: UIState) => state.network.events;
+const getRequests = (state: UIState) => state.network.requests;
+
+export const selectors = {
+  getEvents,
+  getRequests,
+  getNetworkRequests: createSelector(getEvents, getRequests, (events, requests) => ({
+    events,
+    requests,
+  })),
+};
```

Comment:
I think in the past we've exported each selector individually and done `* as selectors` elsewhere. I'm spoiled, but at this point an export object seems like a hassle

Reply:
Yeah, I saw that 😄 . I am ok with sticking with that style for this purpose. In general I stay pretty far away from `*` imports.

Reply:
haha fair enough